IdentityCore/src/cache/metadata/accountMetadata/MSIDAccountMetadataCacheItem.m

```diff
@@ -22,93 +22,53 @@
 // THE SOFTWARE.
 
 #import "MSIDAccountMetadataCacheItem.h"
-#import "MSIDAccountIdentifier.h"
-#import "MSIDAuthority.h"
-#import "MSIDAuthorityFactory.h"
-#import "NSDictionary+MSIDExtensions.h"
+#import "MSIDAccountMetadata.h"
 #import "MSIDAccountMetadataCacheKey.h"
-#import "MSIDRequestParameters.h"
-
-static const NSString *AccountMetadataURLMapKey = @"URLMap";
-
-@interface MSIDAccountMetadataCacheItem()
-@property NSMutableDictionary *internalMap;
-
-@end
 
 @implementation MSIDAccountMetadataCacheItem
+{
+    NSMutableDictionary <NSString *, MSIDAccountMetadata *> *_accountMetadataMap;
+}
 
-- (instancetype)initWithHomeAccountId:(NSString *)homeAccountId
-                             clientId:(NSString *)clientId
-
+- (instancetype)initWithClientId:(NSString *)clientId
 {
-    if (!homeAccountId || !clientId) return nil;
+    if ([NSString msidIsStringNilOrBlank:clientId]) return nil;
```

Comment:
nit: create an error? or at least log?

Reply:
added logging statement here.
Also added parameter check in `MSIDAccountMetadataCacheAccessor` to be more defensive :)